QAchat/python/QAchat.ipynb

```diff
@@ -0,0 +1,428 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# BrainTrust QA Chat Tutorial\n",
+    "\n",
+    "<a target=\"_blank\" href=\"https://colab.research.google.com/github/braintrustdata/braintrust-examples/blob/main/\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>\n",
+    "\n",
+    "Welcome to [BrainTrust](https://www.braintrustdata.com/)! This is a quick tutorial on how to build and evaluate an AI question and answer chat assistant. The assistant answers questions based on the user's information that is already saved in a vector DB (Chroma).\n",
+    "\n",
+    "Before starting, make sure that you have a BrainTrust account. If you do not, please [sign up](https://www.braintrustdata.com) or [get in touch](mailto:info@braintrustdata.com). After this tutorial, learn more by visiting [the docs](http://www.braintrustdata.com/docs)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import braintrust\n",
+    "import chromadb\n",
+    "import openai\n",
+    "from autoevals.string import *\n",
+    "from autoevals.llm import *\n",
+    "\n",
+    "PROJECTNAME = \"QAchatbot\"\n",
+    "OPENAI_API_KEY = \"YOURAPIKEY\"\n",
+    "BT_API_KEY=\"YOURAPIKEY\"\n",
+    "\n",
+    "openai.api_key = OPENAI_API_KEY"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Load in datasets\n",
+    "\n",
+    "First, we'll load two datasets:\n",
+    "1. An evaluation dataset to test our pipeline. This includes input, output pairs like:\n",
+    "```\n",
+    "    {\"input\": \"What is my full name?\", \"output\": \"John Smith. -BT\"}\n",
+    "```\n",
+    "2. A user context dataset to give to the AI assistant as context. This will be stored in a vector db and contains rows like:\n",
+    "```\n",
+    "    {\"category\": \" address\", \"detail\": \"123 Main Street, Anytown, USA\"}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Input, output pairs to evaluate our QAChatbot.\n",
+    "# Upload these to a dataset in BrainTrust so your teammates can also easily use and manage this dataset.\n",
+    "dataset = braintrust.init_dataset(PROJECTNAME, name=\"Basic Evaluation\", api_key=BT_API_KEY)\n",
```

Comment:
we should use an environment variable to prevent passing around te API key as input here. It's generally considered a bad and dangerous practice to pass api keys as plain text arguments in code, but it's also brittle because you need to remember to do it everywhere